src/dex.hs

```diff
@@ -131,34 +132,39 @@ printLitProg JSONDoc prog =
     "{}" -> return ()
     s -> putStrLn s
 
+nonEmptyNewline :: String -> String
 nonEmptyNewline [] = []
 nonEmptyNewline l  = l ++ ['\n']
 
 parseOpts :: ParserInfo CmdOpts
 parseOpts = simpleInfo $ CmdOpts
   <$> parseMode
-  <*> (optional $ strOption $ long "prelude" <> metavar "FILE" <> help "Prelude file")
+  <*> optional (strOption $ long "prelude" <> metavar "FILE" <> help "Prelude file")
   <*> parseEvalOpts
 
+helpOption :: String -> String -> Mod f a
+helpOption optionName options =
```

Comment:
How does this change the printing?

Reply:
Previously, help option strings (`(literate(default)|result-only|HTML|JSON`) where placed on their own line simply because the string was so long and didn't have spaces.

Now, help option strings have an explicitly added line break (`hardline`), allowing us to make the messages themselves nicer by adding spaces. Outer parentheses are also removed.

---

Before:
```console
$ dex --help
Usage: dex COMMAND [--prelude FILE] [--backend ARG] [--logto FILE]

Available options:
  --prelude FILE           Prelude file
  --backend ARG            Backend (LLVM(default)|LLVM-CUDA|interp)
  --logto FILE             File to log to
  -h,--help                Show this help text

$ dex script --help
Usage: dex script FILE [--outfmt ARG] [--allow-errors]

Available options:
  FILE                     Source program
  --outfmt ARG             Output format
                           (literate(default)|result-only|HTML|JSON
  --allow-errors           Evaluate programs containing non-fatal type errors
  -h,--help                Show this help text
```

After:

```console
$ dex --help
Usage: dex COMMAND [--prelude FILE] [--backend ARG] [--logto FILE]

Available options:
  --prelude FILE           Prelude file
  --backend ARG            Backend
                           LLVM (default) | LLVM-CUDA | LLVM-MC | interpreter
  --logto FILE             File to log to
  -h,--help                Show this help text

$ dex script --help
Usage: dex script FILE [--outfmt ARG] [--allow-errors]

Available options:
  FILE                     Source program
  --outfmt ARG             Output format
                           literate (default) | result-only | HTML | JSON
  --allow-errors           Evaluate programs containing non-fatal type errors
  -h,--help                Show this help text
```

Reply:
Quick bikeshed: any opinion on renaming `dex script --outfmt {HTML,JSON}` to `dex script --outfmt {html,json}` for consistent casing? @dougalm

Reply:
Most CLIs seem to use lowercase option values. @apaszke seems supportive of renaming option values to lowercase for consistency: done in https://github.com/google-research/dex-lang/pull/437/commits/37e4f50f555a1966a48c0ed09449639848bb9cf3.

Meta: in general, I don't think it is a good idea to spend too much time bikeshedding - wanted to quickly bounce this change sooner rather than later.